AudioKit/Common/Internals/Helpers/ExceptionCatcher.swift

```diff
@@ -0,0 +1,34 @@
+//
+//  ExceptionCatcher.swift
+//  AudioKit
+//
+//  Created by Stéphane Peter, revision history on Github.
+//  Copyright © 2018 AudioKit. All rights reserved.
+//
+
+import Foundation
+
+public func AKTry(_ operation: @escaping (() throws -> Void)) throws
+{
+    var error: Error?
+    
+    let theTry = {
+        do {
+            try operation()
+        } catch let ex {
+            error = ex
+        }
+    }
+    
+    let theCatch: (NSException) -> Void = { except in
+        error = NSError(domain: "io.audiokit.AudioKit",
```

Comment:
Here's what I was doing for the exception wrapping.  Take it or leave it :)

```
var userInfo = [String: Any]()
userInfo[NSLocalizedDescriptionKey] = exception.description
userInfo[NSLocalizedFailureReasonErrorKey] = exception.reason
userInfo["exception"] = exception
return NSError(domain: "io.audiokit.AudioKit", code: 0, userInfo: userInfo)
```
It would be really cool if we could print the stack trace too.